Compile fail with --enable-heterogeneous

On master, we're getting a compile fail when `--enable-heterogeneous` is specified:

```
make[1]: Entering directory `/home/jsquyres/git/ompi/ompi/mca/coll'
  CC       base/coll_base_alltoall.lo
In file included from ../../../opal/mca/threads/thread_usage.h:32,
                 from ../../../opal/class/opal_object.h:126,
                 from ../../../opal/datatype/opal_datatype.h:44,
                 from ../../../opal/datatype/opal_convertor.h:35,
                 from ../../../ompi/datatype/ompi_datatype.h:40,
                 from base/coll_base_alltoall.c:31:
base/coll_base_alltoall.c: In function ‘mca_coll_base_alltoall_intra_basic_inplace’:
base/coll_base_alltoall.c:81:85: error: invalid use of undefined type ‘struct opal_convertor_master_t’
   81 |         if( OPAL_UNLIKELY(opal_local_arch != ompi_proc->super.proc_convertor->master->remote_arch))  {
      |                                                                                     ^~
../../../opal/include/opal/prefetch.h:44:63: note: in definition of macro ‘OPAL_UNLIKELY’
   44 | #        define OPAL_UNLIKELY(expression) __builtin_expect(!!(expression), 0)
      |                                                               ^~~~~~~~~~
base/coll_base_alltoall.c:83:100: error: invalid use of undefined type ‘struct opal_convertor_master_t’
   83 |                                                             ompi_proc->super.proc_convertor->master->remote_sizes);
      |                                                                                                    ^~
make[1]: *** [base/coll_base_alltoall.lo] Error 1
make[1]: Leaving directory `/home/jsquyres/git/ompi/ompi/mca/coll'
make: *** [all-recursive] Error 1
```

I'm not sure what commit introduced the error.

--- a/ompi/mca/coll/base/coll_base_alltoall.c
+++ b/ompi/mca/coll/base/coll_base_alltoall.c
@@ -29,6 +29,7 @@
 #include "mpi.h"
 #include "ompi/constants.h"
 #include "ompi/datatype/ompi_datatype.h"
+#include "opal/datatype/opal_convertor_internal.h"
 #include "ompi/communicator/communicator.h"
 #include "ompi/mca/coll/coll.h"
 #include "ompi/mca/coll/base/coll_tags.h"

--- a/ompi/mca/coll/base/coll_base_alltoallv.c
+++ b/ompi/mca/coll/base/coll_base_alltoallv.c
@@ -31,6 +31,7 @@
 #include "mpi.h"
 #include "ompi/constants.h"
 #include "ompi/datatype/ompi_datatype.h"
+#include "opal/datatype/opal_convertor_internal.h"
 #include "ompi/communicator/communicator.h"
 #include "ompi/mca/coll/coll.h"
 #include "ompi/mca/coll/base/coll_tags.h"

